library/std/src/io/util.rs

```diff
@@ -8,24 +8,41 @@ use crate::io::{
     self, BufRead, IoSlice, IoSliceMut, Read, ReadBuf, Seek, SeekFrom, SizeHint, Write,
 };
 
-/// A reader which is always at EOF.
+/// A writer which will move data into the void, and a reader which
+/// is always at EOF.
 ///
-/// This struct is generally created by calling [`empty()`]. Please see
-/// the documentation of [`empty()`] for more details.
+/// This struct is generally created by calling [`empty()`]. Please
+/// see the documentation of [`empty()`] for more details.
 #[stable(feature = "rust1", since = "1.0.0")]
 #[non_exhaustive]
-#[derive(Copy, Clone, Default)]
+#[derive(Copy, Clone, Debug, Default)]
 pub struct Empty;
 
-/// Constructs a new handle to an empty reader.
+/// Creates an instance of a writer which will successfully consume all data, as
```

Comment:
I think similarly to my suggestion on Empty, I would rephrase this to more directly talk about a single "thing" that implements two traits and describe that behavior.

Otherwise, this to me sounds like we're returning a writer and a reader (e.g., `fn empty() -> (impl Write, impl Read)` or something), which isn't the case.

Reply:
Ping @vidhanio. I agree. The wording is a little weird here. Perhaps something like:

> Creates a value that is always at EOF for reads, and ignores all data written.
>
> The value returned implements both the [`Read`] and [`Write`] traits.